plutus-metatheory/src/Builtin.lagda.md

```diff
@@ -79,7 +95,127 @@ data Builtin : Set where
+    signature divideInteger                   = ∙ [ integer , integer ]⟶ integer
+    signature quotientInteger                 = ∙ [ integer , integer ]⟶ integer
+    signature remainderInteger                = ∙ [ integer , integer ]⟶ integer
+    signature modInteger                      = ∙ [ integer , integer ]⟶ integer
+    signature equalsInteger                   = ∙ [ integer , integer ]⟶ bool
+    signature lessThanInteger                 = ∙ [ integer , integer ]⟶ bool
+    signature lessThanEqualsInteger           = ∙ [ integer , integer ]⟶ bool
+    signature appendByteString                = ∙ [ bytestring , bytestring ]⟶ bytestring
+    signature consByteString                  = ∙ [ integer , bytestring ]⟶ bytestring
+    signature sliceByteString                 = ∙ [ integer , integer , bytestring ]⟶ bytestring                            
+    signature lengthOfByteString              = ∙ [ bytestring ]⟶ integer
+    signature indexByteString                 = ∙ [ bytestring , integer ]⟶ integer
+    signature equalsByteString                = ∙ [ bytestring , bytestring ]⟶ bool
+    signature lessThanByteString              = ∙ [ bytestring , bytestring ]⟶ bool
+    signature lessThanEqualsByteString        = ∙ [ bytestring , bytestring ]⟶ bool
+    signature sha2-256                        = ∙ [ bytestring ]⟶ bytestring
+    signature sha3-256                        = ∙ [ bytestring ]⟶ bytestring
+    signature blake2b-256                     = ∙ [ bytestring ]⟶ bytestring
+    signature verifyEd25519Signature          = ∙ [ bytestring , bytestring , bytestring ]⟶ bool
+    signature verifyEcdsaSecp256k1Signature   = ∙ [ bytestring , bytestring , bytestring ]⟶ bool
+    signature verifySchnorrSecp256k1Signature = ∙ [ bytestring , bytestring , bytestring ]⟶ bool
+    signature appendString                    = ∙ [ string , string ]⟶ string
+    signature equalsString                    = ∙ [ string , string ]⟶ bool
+    signature encodeUtf8                      = ∙ [ string ]⟶ bytestring
+    signature decodeUtf8                      = ∙ [ bytestring ]⟶ string 
+    signature ifThenElse                      = ∀a [ bool , a , a ]⟶ a
+    signature chooseUnit                      = ∀a [ a , unit ]⟶ a
+    signature trace                           = ∀a [ string , a ]⟶ a
+    signature fstPair                         = ∀b,a [ pair b a ]⟶ b
+    signature sndPair                         = ∀b,a [ pair b a ]⟶ a
+    signature chooseList                      = ∀b,a [ list b , a , a ]⟶ a
+    signature mkCons                          = ∀a [ a , list a ]⟶ list a
+    signature headList                        = ∀a [ list a ]⟶ a
+    signature tailList                        = ∀a [ list a ]⟶ list a
+    signature nullList                        = ∀a [ list a ]⟶ bool
+    signature chooseData                      = ∀a [ Data , a , a , a , a , a ]⟶ a
+    signature constrData                      = ∙ [ integer , list Data ]⟶ Data
```

Comment:
Is `Data` the only one that starts with a capital letter? Seems a bit inconsistent.

Reply:
It's awkward as data is a keyword

Reply:
Using the word datum might be an option

Reply:
yes, or something like `pdata` (for Plutus data)

Reply:
I have renamed the different constructors named `Data` to `pdata`. Now it will be easier to rename as it doesn't clash with module names.